### DISCUSSION TITLE
How to plot Electrostatic Potential?

### DISCUSSION THREAD

**Discussion by icamps** [grimme-lab/xtb#547] [Q&A]
Hello,

I wonder how can I plot the electrostatic potential calculated with xTB.

I have three files: `cosmo`, `dat` and [profile](url) but don't know how to play with them.

Thanks in advance.

Camps

**Comment by awvwgk** [comment]
Could you provide some information on the calculation you performed here, like the command line options or the detailed input file?

For the ESP calculation the output is usually on a Lebedev angular grid rather than on a cube grid due to the computational costs associated with the ESP calculation, the `dat` file contains the positions of the surface elements and the ESP value and would be the most useful output for visualization.

**Reply by rlavendomme** [reply]
Hello,

I think the main issue here is that ESP output files with .cosmo and .dat extensions are not natively recognized by most common visualization programs such as VMD, PyMOL, ChimeraX, jmol, Avogadro.

I would suggest to indicate in [xtb documentation](https://xtb-docs.readthedocs.io/en/latest/index.html) what is the format of the ESP surface output and recommend programs to visualize/plot ESP surfaces. Alternatively, a method to convert output .cosmo/.dat files to another format more commonly used in chemistry would also work.

**Comment by icamps** [comment]
@awvwgk I just used the following:

Command line: `xtb --input in_xtb.inp Glifosato.xyz --molden --pop`
Input file `in_xtb.inp`:

```
$scc
   temp=0
$write
   output file=properties.out
   json=true
   esp=true
   density=true
   spin population=true
   spin density=true
   mos=true
   wiberg=true
   charges=true
   mulliken=false

```

**Reply by giberger** [reply]
Hello everyone,

I have the exact same question, so digging out this old thread 🤓

How to plot the esp with the dat and cosmo files?

Thanks a lot
Gilles

**Comment by icamps** [comment]
Any updates about this issue?
